### PR TITLE
Add support for processing DefaultValueAttribute in OpenApi schemas

### LINKED ISSUE #56215: Properties with the [DefaultValue] attribute should have a default value in the generated OpenAPI document
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

A property of a class or record that has the [DefaultValue] attribute currently does not have a `default` in the schema produced by the .NET 9 OpenAPI document generation feature.

### Expected Behavior

A property with the [DefaultValue] attribute should have a `default` with the value specified in the generated OpenAPI schema.

### Steps To Reproduce

Not minimal, but there is a recreate in the "Schemas" section of my [aspnet-openapi-examples](https://github.com/mikekistler/aspnet-openapi-examples) project.

### Exceptions (if any)

_No response_

### .NET Version

9.0-preview4

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by captainsafia**
Closes https://github.com/dotnet/aspnetcore/issues/56215.

### CODE REVIEW

File: src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.ParameterSchemas.cs [hunk 1]
```diff
@@ -154,6 +155,12 @@ public async Task GetOpenApiParameters_HandlesRouteParameterWithRouteConstraint(
...
         // Default value for enums is serialized as string when a converter is registered.
         [(Status status = Status.Pending) => { }, (IOpenApiAny defaultValue) => Assert.Equal("Pending", ((OpenApiString)defaultValue).Value)],
+        [([DefaultValue(2)] int id) => { }, (IOpenApiAny defaultValue) => Assert.Equal(2, ((OpenApiInteger)defaultValue).Value)],
+        [([DefaultValue(3f)] float id) => { }, (IOpenApiAny defaultValue) => Assert.Equal(3, ((OpenApiInteger)defaultValue).Value)],
+        [([DefaultValue("test")] string id) => { }, (IOpenApiAny defaultValue) => Assert.Equal("test", ((OpenApiString)defaultValue).Value)],
+        [([DefaultValue(true)] bool id) => { }, (IOpenApiAny defaultValue) => Assert.True(((OpenApiBoolean)defaultValue).Value)],
+        [([DefaultValue(TaskStatus.Canceled)] TaskStatus status) => { }, (IOpenApiAny defaultValue) => Assert.Equal(6, ((OpenApiInteger)defaultValue).Value)],
+        [([DefaultValue(Status.Pending)] Status status) => { }, (IOpenApiAny defaultValue) => Assert.Equal("Pending", ((OpenApiString)defaultValue).Value)],
```
Comment (martincostello) [inline src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.ParameterSchemas.cs:163]: Based on some [recent bugs](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues?q=is%3Aissue+defaultvalue+is%3Aopen+created%3A%3E2024-05-01+) we've had in Swashbuckle, you might want to add some test cases for:

- `[DefaultValue(null)] int? id`
- `[DefaultValue(2)] int? id`
- `[DefaultValue(null)] string? id`
- `[DefaultValue("foo")] string? id`
- `[DefaultValue(null)] TaskStatus? status`
- `[DefaultValue(TaskStatus.Canceled)] TaskStatus? status`

Comment (captainsafia) [reply src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.ParameterSchemas.cs:163]: Great list! Thank you for sharing. I'll add these. I suspect that `[DefaultValue(null)]` will be interesting to sus out.

File: src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs [hunk 2]
```diff
@@ -71,7 +76,7 @@ internal async Task<OpenApiSchema> GetOrCreateSchemaAsync(Type type, ApiParamete
         var schemaAsJsonObject = _schemaStore.GetOrAdd(key, CreateSchema);
         if (parameterDescription is not null)
         {
-            schemaAsJsonObject.ApplyParameterInfo(parameterDescription);
+            schemaAsJsonObject.ApplyParameterInfo(parameterDescription, OpenApiJsonSchemaContext.Default.GetTypeInfo(type)!);
```
Comment (BrennanConroy) [inline src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs:79]: Could you explain why this is always non-null?

Comment (captainsafia) [reply src/OpenApi/src/Services/Schemas/OpenApiSchemaService.cs:79]: 🤦🏽‍♀️ Good catch -- goof on my part.

**Review by BrennanConroy — APPROVED** [review]

**Pull request merged**